components/server/src/workspace/workspace-starter.ts

```diff
@@ -345,33 +346,48 @@ export class WorkspaceStarter {
                 auth.setSelective(selectiveAuth);
             }
             if (WorkspaceImageSourceDocker.is(imgsrc)) {
-                // We only build Docker images from Git initializer. To ensure that createGitInitializer relies on the
-                // unset "ref". To double down on this, we do modify the initializer request. We're just using createGitInitializer
-                // for the authentication handling.
-                const { git, disposable } = await this.createGitInitializer({span}, workspace, {
-                    ...imgsrc.dockerFileSource,
-                    title: "irrelevant",
-                    ref: undefined,
-                }, user);
-                git.setCloneTaget(imgsrc.dockerFileSource.revision);
-                git.setTargetMode(CloneTargetMode.REMOTE_COMMIT);
-                git.setCheckoutLocation(".")
-                const source = new WorkspaceInitializer();
-                source.setGit(git);
+                let source: WorkspaceInitializer;
+                const disp = new DisposableCollection();
+                let checkoutLocation = this.getCheckoutLocation(workspace);
+                if (!imgsrc.dockerFileSource) {
+                    const {initializer, disposable} = await this.createInitializer({span}, workspace, workspace.context, user);
+                    source = initializer;
+                    disp.push(disposable);
+                } else {
+                    //TODO we cannot change this initializer structure now because it is part of how baserefs are computed in image-builder.
+                    // Image builds should however just use the initialization if the workspace they are running for (i.e. the one from above).
+                    const { git, disposable } = await this.createGitInitializer({span}, workspace, {
+                        ...imgsrc.dockerFileSource,
+                        title: "irrelevant",
+                        ref: undefined,
+                    }, user);
+                    disp.push(disposable);
+                    git.setCloneTaget(imgsrc.dockerFileSource.revision);
+                    git.setTargetMode(CloneTargetMode.REMOTE_COMMIT);
+                    checkoutLocation = "."
+                    git.setCheckoutLocation(checkoutLocation);
+                    source = new WorkspaceInitializer();
+                    source.setGit(git);
+                }
 
                 // Dockerfile and context path are both relative to the checkout location
-                let contextPath = (workspace.config.image as ImageConfigFile).context || ".";
-                let dockerFilePath = imgsrc.dockerFilePath;
+                let contextPath = checkoutLocation;
+                let dockerFilePath = checkoutLocation + '/' + imgsrc.dockerFilePath;
```

Comment:
This seems to be at odds with the comment above. Either the comment or the code is wrong :)